src/enrichment/models/cider.rs

```diff
@@ -150,4 +128,270 @@ impl CiderImagesModel {
+        let outputs = self.model.run(model_inputs)?;
+        let (_, scores) = outputs["logits"]
+            .try_extract_tensor::<f32>()
+            .map_err(|_| ModelError::ModelOutputToVecError)?;
+
+        let vec_scores = scores.to_vec();
+        let temp_array = Array::from_shape_vec((1, 5), vec_scores)?;
+        let output = Self::softmax(temp_array);
+        Ok(output.row(0).to_vec())
+    }
+
+    #[instrument(skip_all, err)]
+    pub fn photometry_inputs(
+        &self,
+        photometry: Vec<PhotometryMag>,
+    ) -> Result<
+        (
+            ArrayBase<OwnedRepr<f32>, Dim<[usize; 3]>>,
+            ArrayBase<OwnedRepr<bool>, Dim<[usize; 2]>>,
+        ),
+        ModelError,
+    > {
+        let mut sorted_photometry = Vec::from(photometry);
+        sorted_photometry.sort_by(|a, b| a.time.partial_cmp(&b.time).unwrap());
+        sorted_photometry.dedup_by(|a, b| a.time == b.time && a.band == b.band);
+        let mut result = Vec::new();
+
+        let mut bands: std::collections::HashMap<String, Vec<PhotometryMag>> =
+            std::collections::HashMap::new();
+        for point in sorted_photometry {
+            bands
+                .entry(point.band.clone())
+                .or_insert_with(Vec::new)
+                .push(point);
+        }
+        // Process each band separately
+        for (_band, mut points) in bands {
+            // Sort by time
+            points.sort_by(|a, b| a.time.partial_cmp(&b.time).unwrap());
+            let mut i = 0;
+            while i < points.len() {
+                let mut group = vec![&points[i]];
+                let mut j = i + 1;
+
+                // Collect all points within time_threshold of the first point in group
+                while j < points.len() && (points[j].time - points[i].time) <= 0.5 {
+                    group.push(&points[j]);
+                    j += 1;
+                }
+
+                // Combine the group
+                let combined = Self::combine_group(&group);
+                result.push(combined);
+
+                i = j;
+            }
+        }
+        // Sort final result by time
+        result.sort_by(|a, b| a.time.partial_cmp(&b.time).unwrap());
+
+        let initial_t = result[0].time;
+        let mut dt = Vec::new();
+        let mut delta_t = Vec::new();
+        let mut temp = result[0].time;
+        let mut log_fluxes = Vec::new();
+        let mut sigma_fluxes = Vec::new();
+        let mut tail1: Vec<f32> = Vec::new();
+        let mut tail2: Vec<f32> = Vec::new();
+        let mut tail3: Vec<f32> = Vec::new();
+
+        for point in &result {
+            let diff_delta_t = point.time - initial_t;
+            let diff_dt = point.time - temp;
+            temp = point.time;
+            dt.push(diff_dt as f32);
+            delta_t.push(diff_delta_t as f32);
+            log_fluxes.push(point.mag.max(1e-6).log10() as f32);
+            sigma_fluxes.push(point.mag_err / (point.mag * 2.302585) as f32);
```

Comment:
all of these normalizing factors should be documented somewhere. Would be good to have a document (if not here, in a cider repo) that lists them, to serve as a source.